mcode/mcode-coverage-checker
# Past Files Table

# Description
Addresses [STEAM 977](https://jira.mitre.org/browse/STEAM-977), adding a Past Files component. This component includes a table that displays the file name, file size, when the file was added, and a JSON viewer for uploaded mCODE bundles.

## (Feature) Demo/Screenshots
Table:

https://user-images.githubusercontent.com/53443041/198703408-0c66055d-be85-49da-bac8-93557373c838.mov



Reference from Figma:
<img width="1655" alt="Screen Shot 2022-10-26 at 4 37 12 PM" src="https://user-images.githubusercontent.com/53443041/198701572-68dd4acf-f33a-4ae9-9f58-eb33bb44652d.png">

## Important Changes
1. Adds a table to `PastFiles.js` capable of displaying metadata of uploaded JSON files, viewing these files, and deleting these files. Note that these files are hardcoded right now, but should be passed as props once this component is actually hooked into the rest of the web app.
2. Adds the `PastFiles` component to `App.js` for easy testing. 

 
## Testing Recommendations
1. Ensure that table styling is consistent with the Style Guide, Figma reference, and generally is appealing to the user. 
2. Ensure that code structure is sensible and maintainable. I tried to use tailwind classes for styling where possible, but in instances where the Style Guide specified a font size value that was not included a tailwind class, I included it as custom style. If you see anything that could or should be styled as a tailwind class rather than a custom style, please point it out! 
3. Ensure file deletion and toggling the JSON viewer works as expected. 
# Checklists

**Submitter:**

@ACCT1 :
